### ISSUE TITLE
Query: reuse include pipeline for queries projecting uncomposed collection navigations

### ISSUE THREAD

**Issue by maumar** [aspnet/EntityFramework#8584]
```
ctx.Customers.Select(c => c.Orders)
```

currently produces N+1 queries, however, similar query:

```
ctx.Customers.Include(c => c.Orders)
```

is much better, producing only 2 queries.

We should be able to leverage the include pipeline for the first query

**Comment by maumar** [comment]
related to #8229

**Comment by maumar** [comment]
fixed in 0075cb7c831bb3618bee4a84c9bfa86a499ddc6c
